### ISSUE TITLE
Failed to compile, can't resolve dom-helpers modules

### ISSUE THREAD

**Issue by heregoesthekrr** [doczjs/docz#1102]
Got docz installed on existing UI library, trying to run docz dev with no mdx files at all

Got this error while bundling:
```shell⠀
ERROR  Failed to compile with 5 errors                                                                                                                                                                                                                  10:55:41 AM

These dependencies were not found:
⠀
* dom-helpers/events/off in ../node_modules/scroll-behavior/lib/index.js
* dom-helpers/events/on in ../node_modules/scroll-behavior/lib/index.js
* dom-helpers/query/scrollLeft in ../node_modules/scroll-behavior/lib/index.js
* dom-helpers/query/scrollTop in ../node_modules/scroll-behavior/lib/index.js
* dom-helpers/util/requestAnimationFrame in ../node_modules/scroll-behavior/lib/index.js
```

Seems that it is some kind of versioning issue: node_modules in ./.docz contains dom-helpers v5,  which has no such exports unlike v3 which probably should be resolved and installed

**Comment by rakannimer** [comment]
Hey @barbalex 

I think you're right, the problem is from conflicting required versions of `dom-helpers` with breaking changes between them 😓 

You could maybe try using an older version of `@material-ui/core` to be compatible with your `graphql-voyager#@material-ui/core` dependency. 

That would make all dependencies to dom-helpers ask for version 3.4.0

Or you could submit PRs to : 

- react-swipeable-views and update dom-helpers dependency
- graphql-voyager and update material-ui/core dependency
- scroll-behavior to update dom-helpers version and propagate it to gatsby-react-router-scroll and then upstream to gatsby.

**Comment by barbalex** [comment]
@rakannimer Thanks, I am trying to do that, see https://github.com/react-bootstrap/dom-helpers/issues/51#issuecomment-531189070 and following comments.

I guess this problem has nothing to do with docz itself so the issue could be closed.

**Comment by TrySound** [comment]
@rakannimer I don't see anything wrong with dependencies. Conflicting ones are nested in packages properly. But for some reason docz resolver finds only dom-helpers from the root.
https://github.com/react-bootstrap/dom-helpers/issues/51#issuecomment-531236558

**Comment by rakannimer** [comment]
@TrySound Thanks ! I was following along the conversation https://github.com/react-bootstrap/dom-helpers/issues/51 

I'm honestly not sure why that would happen. Docz uses Gatsby as a bundler, any idea how I can find out if it's a Gatsby or a docz problem ?

**Comment by artem-popov** [comment]
I have fixed that by adding 
  "resolutions": {
    "dom-helpers": "3.4.0"
  },

**Comment by rahul3103** [comment]
I am also facing same issue with `next-create-app`
With `material-ui` and `docz` together.

. Then I tried using  npx create-docz-app my-docz-app. And added `yarn add @material-ui/core`
and I am facing same `dom-helper` issue.

```
yarn run dev     
yarn run v1.19.1
warning ../package.json: No license field
$ docz dev

Building app
warn Configuring yargs through package.json is deprecated and will be removed in the next major release, please use the JS API instead.
success open and validate gatsby-configs - 1.260s
success load plugins - 0.671s
success onPreInit - 0.004s
info One or more of your plugins have changed since the last time you ran Gatsby. As
a precaution, we're deleting your site's cache to ensure there's no stale data.
success initialize cache - 0.031s
success copy gatsby files - 0.107s
success onPreBootstrap - 0.327s
warn Configuring yargs through package.json is deprecated and will be removed in the next major release, please use the JS API instead.
success source and transform nodes - 0.748s
success building schema - 0.315s
success createPages - 0.136s
success createPagesStatefully - 0.066s
success onPreExtractQueries - 0.003s
success update schema - 0.035s
success extract queries from components - 0.089s
success write out requires - 0.010s
success write out redirect data - 0.003s
success onPostBootstrap - 0.002s
⠀
info bootstrap finished - 8.123 s
⠀
success run queries - 0.080s - 7/7 87.15/s
warn ESLintError: 
/Users/rahulshrivastava/projects/personal/my-docz-app/src/components/Alert.jsx
  18:7  warning  Duplicate key 'background'  no-dupe-keys

✖ 1 problem (0 errors, 1 warning)


 ERROR #98123  WEBPACK

Generating development JavaScript bundle failed

Can't resolve 'dom-helpers/events/off' in '/Users/rahulshrivastava/projects/personal/my-docz-app/node_modules/scroll-behavior/lib'

File: ../node_modules/scroll-behavior/lib/index.js


 ERROR #98123  WEBPACK

Generating development JavaScript bundle failed

Can't resolve 'dom-helpers/events/on' in '/Users/rahulshrivastava/projects/personal/my-docz-app/node_modules/scroll-behavior/lib'

File: ../node_modules/scroll-behavior/lib/index.js


 ERROR #98123  WEBPACK

Generating development JavaScript bundle failed

Can't resolve 'dom-helpers/query/scrollLeft' in '/Users/rahulshrivastava/projects/personal/my-docz-app/node_modules/scroll-behavior/lib'

File: ../node_modules/scroll-behavior/lib/index.js


 ERROR #98123  WEBPACK

Generating development JavaScript bundle failed

Can't resolve 'dom-helpers/query/scrollTop' in '/Users/rahulshrivastava/projects/personal/my-docz-app/node_modules/scroll-behavior/lib'

File: ../node_modules/scroll-behavior/lib/index.js


 ERROR #98123  WEBPACK

Generating development JavaScript bundle failed

Can't resolve 'dom-helpers/util/requestAnimationFrame' in '/Users/rahulshrivastava/projects/personal/my-docz-app/node_modules/scroll-behavior/lib'

File: ../node_modules/scroll-behavior/lib/index.js

failed Building development bundle - 6.386s


———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————
6 pages                                                                       Failed                                                                       docz-app

App ready on http://localhost:3001
```

**Comment by GabeCabrera** [comment]
Bump of Rahuls comment, I am also receiving the same issue using `create-react-app`, `material-ui/core` and `docz` together.

```
Generating development JavaScript bundle failed

Can't resolve 'dom-helpers/util/requestAnimationFrame' in '/Users/gabecabrera/Desktop/code/bobber-fullstack/bobber-ui/node_modules/scroll-behavior/lib'

File: ../node_modules/scroll-behavior/lib/index.js

failed Building development bundle - 8.112s
```

**Comment by aleksandervalle** [comment]
Installed material-ui/core on a pretty fresh docz-install, and getting same error -- conflicting versions of dom-helpers.

After reading https://github.com/webpack/webpack/issues/6538, I tried commenting out this whole if-statement (https://github.com/doczjs/docz/blob/master/core/gatsby-theme-docz/lib/onCreateWebpackConfig.js#L23):

```
  if (hasParentNodeModules) {
    actions.setWebpackConfig({
      resolve: {
        modules: [nodeModules, parentNodeModules],
      },
    })
  }
```

I'll admit ignorance to webpack module resolution, but seems to me that things are resolving fine, i.e. things seem to be working.

**Comment by aleksandervalle** [comment]
@rakannimer Superb. Docz is fantastic, thanks for doing an awesome job! :-)

**Comment by rakannimer** [comment]
Thanks @aleksandervalle :)

The issue should be fixed in `2.0.0-rc.77` 👍 

Going to close this issue, please comment here if the problem re-occurs.

**Comment by uxitten** [comment]
I resolved

```bash
rm -rf node_modules
rm package-lock.json
npm i
```
